Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -0,0 +1,232 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading.Tasks;
+using Jellyfin.Data.Entities;
+using Jellyfin.Data.Entities.Security;
+using Jellyfin.Data.Enums;
+using Jellyfin.Data.Events;
+using Jellyfin.Data.Queries;
+using MediaBrowser.Controller.Devices;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Model.Devices;
+using MediaBrowser.Model.Querying;
+using MediaBrowser.Model.Session;
+using Microsoft.EntityFrameworkCore;
+
+namespace Jellyfin.Server.Implementations.Devices
+{
+    /// <summary>
+    /// Manages the creation, updating, and retrieval of devices.
+    /// </summary>
+    public class DeviceManager : IDeviceManager
+    {
+        private readonly JellyfinDbProvider _dbProvider;
+        private readonly IUserManager _userManager;
+        private readonly ConcurrentDictionary<string, ClientCapabilities> _capabilitiesMap = new ();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeviceManager"/> class.
+        /// </summary>
+        /// <param name="dbProvider">The database provider.</param>
+        /// <param name="userManager">The user manager.</param>
+        public DeviceManager(JellyfinDbProvider dbProvider, IUserManager userManager)
+        {
+            _dbProvider = dbProvider;
+            _userManager = userManager;
+        }
+
+        /// <inheritdoc />
+        public event EventHandler<GenericEventArgs<Tuple<string, DeviceOptions>>>? DeviceOptionsUpdated;
+
+        /// <inheritdoc />
+        public void SaveCapabilities(string deviceId, ClientCapabilities capabilities)
+        {
+            _capabilitiesMap[deviceId] = capabilities;
+        }
+
+        /// <inheritdoc />
+        public async Task UpdateDeviceOptions(string deviceId, DeviceOptions options)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            await dbContext.Database
+                .ExecuteSqlInterpolatedAsync($"UPDATE [DeviceOptions] SET [CustomName] = {options.CustomName}")
```

Comment:
Won't this update ALL `DeviceOptions` to have the specified `CustomName`?
Also, why use raw SQL here (and why not a parameterized query if raw sql)?

Reply:
Yes, this was just me playing around with bulk updates and forgetting to go back and actually implement the method